pype/plugins/global/publish/extract_review.py

```diff
@@ -13,22 +19,1064 @@ class ExtractReview(pyblish.api.InstancePlugin):
+        if output_ext is None:
+            output_ext = os.path.splitext(full_input_path)[1]
+
+        # TODO Define if extension should have dot or not
+        if output_ext.startswith("."):
+            output_ext = output_ext[1:]
+
+        # Store extension to representation
+        new_repre["ext"] = output_ext
+
+        self.log.debug("New representation ext: `{}`".format(output_ext))
+
+        # Output is image file sequence witht frames
+        output_is_sequence = (
+            (output_ext in self.image_exts)
+            and "sequence" in output_def["tags"]
+        )
+
+        if output_is_sequence:
+            new_repre_files = []
+            frame_start = temp_data["output_frame_start"]
+            frame_end = temp_data["output_frame_end"]
+
+            filename_base = "{}_{}".format(filename, filename_suffix)
+            # Temporary tempalte for frame filling. Example output:
+            # "basename.%04d.mov" when `frame_end` == 1001
+            repr_file = "{}.%{:0>2}d.{}".format(
+                filename_base, len(str(frame_end)), output_ext
+            )
+
+            for frame in range(frame_start, frame_end + 1):
+                new_repre_files.append(repr_file % frame)
+
+            new_repre["sequence_file"] = repr_file
+            full_output_path = os.path.join(
+                staging_dir, filename_base, repr_file
+            )
+
+        else:
+            repr_file = "{}_{}.{}".format(
+                filename, filename_suffix, output_ext
+            )
+            full_output_path = os.path.join(staging_dir, repr_file)
+            new_repre_files = repr_file
+
+        # Store files to representation
+        new_repre["files"] = new_repre_files
+
+        # Make sure stagingDire exists
+        staging_dir = os.path.normpath(os.path.dirname(full_output_path))
+        if not os.path.exists(staging_dir):
+            self.log.debug("Creating dir: {}".format(staging_dir))
+            os.makedirs(staging_dir)
+
+        # Store stagingDir to representaion
+        new_repre["stagingDir"] = staging_dir
+
+        # Store paths to temp data
+        temp_data["full_input_path"] = full_input_path
+        temp_data["full_input_path_single_file"] = full_input_path_single_file
+        temp_data["full_output_path"] = full_output_path
+
+        self.log.debug("Input path {}".format(full_input_path))
+        self.log.debug("Output path {}".format(full_output_path))
+
+        return full_input_path, full_output_path
+
+    def audio_args(self, instance, temp_data):
+        """Prepares FFMpeg arguments for audio inputs."""
+        audio_in_args = []
+        audio_filters = []
+        audio_out_args = []
+        audio_inputs = instance.data.get("audio")
+        if not audio_inputs:
+            return audio_in_args, audio_filters, audio_out_args
+
+        for audio in audio_inputs:
+            # NOTE modified, always was expected "frameStartFtrack" which is
```

Comment:
Audio inputs always expected "frameStartFtrack" key in instance. I think it's not OK...

Reply:
definitely not ok.